bcda/auth/plugin/alpha.go

```diff
@@ -1,15 +1,60 @@
 package auth
 
 import (
+	"encoding/json"
 	"errors"
+	"github.com/CMSgov/bcda-app/bcda/auth"
+	"github.com/pborman/uuid"
+	"regexp"
+	"time"
 
-	jwt "github.com/dgrijalva/jwt-go"
+	"github.com/CMSgov/bcda-app/bcda/database"
+	"github.com/CMSgov/bcda-app/bcda/models"
+	"github.com/dgrijalva/jwt-go"
 )
 
 type AlphaAuthPlugin struct{}
 
+// This implementation expects one value in params, an id the API knows this client by in string form
+// It returns a single string as well, being the clientID this implementation knows this client by
+// NB: Other implementations will probably expect more input, and will certainly return more data
 func (p *AlphaAuthPlugin) RegisterClient(params []byte) ([]byte, error) {
-	return nil, errors.New("Not yet implemented")
+	var (
+		empty []byte
+		err   error
+		j     interface{}
+	)
+
+	if err = json.Unmarshal(params, &j); err != nil {
+		return empty, err
+	}
+	paramsMap := j.(map[string]interface{})
+
+	acoUUID, ok := paramsMap["clientID"].(string)
+	if !ok {
+		return empty, errors.New("missing or otherwise invalid clientID")
+	}
+
+	// We'll check carefully in this method, because we're returning something to be used as an id
+	// Normally, a plugin would treat this value as a black box external key, but this implementation is
+	// intimate with the API. So, we're going to protect against accidental bad things
+	if acoUUID == "" || len(acoUUID) > 36 {
```

Comment:
Do we need this when we have the regex matcher right after it?

Reply:
Conventional wisdom is that regex is expensive, so always make it the last check, after you've ruled out other failures. I also like the documentation aspect of length checking the string, particularly when length is a requirement of the format.